### ISSUE TITLE
calls such as MNode.get, MNode.create() can only use the default certificate name

### ISSUE THREAD

**Issue by sbpcs59** [DataONEorg/rdataone#58]
Allow specifying alternate location for the certificate, i.e. add a slot D1Node@certificate that is initialized 
when MNode or CNode objects are created.

**Comment by lauren-walker** [comment]
The problem seems to be that in the MNode create() function, a new instance of CertificateManager is created, overriding any custom attributes such as "location". (line 259 of MNode.R). Perhaps a CM can be optionally passed to the create() function (and others?) so that a custom location can be used.

**Comment by mbjones** [comment]
Yeah, CertificateManager used to be a singleton when it was a Java class, so we'll need to redesign to determine how to handle this.  We can either: 1) turn CM back into a singleton, or 2) allow an optional cm CertificateManager to be passed to all methods that utilize authentication.  The first option is simpler for users to call, but doesn't allow multiple certificate manager instances (which would be rarely needed).  The second option requires more extensive code changes.

**Comment by mbjones** [comment]
Here's an example Singleton implementation in R:
  https://gist.github.com/jverzani/1953641

**Comment by leinfelder** [comment]
Not sure why you'd need multiple instances of a certificate manager. The CertificateManager singleton in Java still allows you to use it with multiple identities - you just need to register each DN with its corresponding cert/key pair so that the CertificateManager singleton can�drumroll�manage client certificates and make requests as any of those DNs when asked to do so. In Java, we inspect the Session.subject to determine which certificate/key to send alonf with any of the API requests.
